internal/api/utils.go

```diff
@@ -404,7 +455,25 @@ func filterNotifications(recommendationID string, clusterUUID string, recommenda
 	return recommendationJSON
 }
 
-func UpdateRecommendationJSON(recommendationID string, clusterUUID string, jsonData datatypes.JSON) map[string]interface{} {
+func dropBoxPlotsObject(recommendationJSON map[string]interface{}) map[string]interface{} {
+
+	recommendation_terms, ok := recommendationJSON["recommendation_terms"].(map[string]interface{})
+	if !ok {
+		log.Error("recommendation data not found in JSON")
+		return recommendationJSON
+	}
+
+	for _, period := range []string{"short_term", "medium_term", "long_term"} {
+		intervalData, ok := recommendation_terms[period].(map[string]interface{})
+		if !ok {
+			continue
+		}
+		delete(intervalData, "plots")
+	}
+	return recommendationJSON
+}
```

Comment:
We can leverage Postgresql capabilities here, it took me sometime however there is way to exclude the plots data in select statement itself,
~~~
# select recommendations #- '{recommendation_terms,long_term,plots}' #- '{recommendation_terms,medium_term,plots}' #- '{recommendation_terms,short_term,plots}' from recommendation_sets;
~~~

On the top of this we can create a view so recommendation-set only queries from that view,
~~~
# create view recommendation_sets_no_box_plots as select id, workload_id, container_name, monitoring_start_time, monitoring_end_time, recommendations #- '{recommendation_terms,long_term,plots}' #- '{recommendation_terms,medium_term,plots}' #- '{recommendation_terms,short_term,plots}' as recommendations from recommendation_sets;
~~~

Locally I tested this with notifications as its present in all three terms and it works,
~~~
# select recommendations #- '{recommendation_terms,long_term,notifications}' #- '{recommendation_terms,medium_term,notifications}' #- '{recommendation_terms,short_term,notifications}' from recommendation_sets;
~~~

Reply:
As discussed, let's take this up in a separate iteration.